main.go

```diff
@@ -55,175 +48,157 @@ func main() {
 	flag.Parse()
 
-	if !operator {
-		runController(kubeconfig, masterURL)
-	} else {
-		runOperator(kubeconfig, masterURL)
-	}
-}
+	sha, release := version.GetReleaseInfo()
+	log.Printf("Version: %s\tcommit: %s\n", release, sha)
 
-// runController runs the faas-netes imperative controller
-func runController(kubeconfig, masterURL string) {
 	clientCmdConfig, err := clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
 	if err != nil {
 		log.Fatalf("Error building kubeconfig: %s", err.Error())
 	}
 
-	clientset, err := kubernetes.NewForConfig(clientCmdConfig)
+	kubeClient, err := kubernetes.NewForConfig(clientCmdConfig)
 	if err != nil {
 		log.Fatalf("Error building Kubernetes clientset: %s", err.Error())
-
 	}
 
-	functionNamespace := "default"
-
-	if namespace, exists := os.LookupEnv("function_namespace"); exists {
-		functionNamespace = namespace
+	faasClient, err := clientset.NewForConfig(clientCmdConfig)
+	if err != nil {
+		log.Fatalf("Error building OpenFaaS clientset: %s", err.Error())
 	}
 
 	readConfig := types.ReadConfig{}
 	osEnv := providertypes.OsEnv{}
-	cfg, err := readConfig.Read(osEnv)
+	config, err := readConfig.Read(osEnv)
+
 	if err != nil {
 		log.Fatalf("Error reading config: %s", err.Error())
 	}
 
-	log.Printf("HTTP Read Timeout: %s\n", cfg.FaaSConfig.GetReadTimeout())
-	log.Printf("HTTP Write Timeout: %s\n", cfg.FaaSConfig.WriteTimeout)
-	log.Printf("HTTPProbe: %v\n", cfg.HTTPProbe)
-	log.Printf("SetNonRootUser: %v\n", cfg.SetNonRootUser)
-
-	sha, release := version.GetReleaseInfo()
-	log.Printf("Starting operator. Version: %s\tcommit: %s", release, sha)
-
 	deployConfig := k8s.DeploymentConfig{
 		RuntimeHTTPPort: 8080,
-		HTTPProbe:       cfg.HTTPProbe,
-		SetNonRootUser:  cfg.SetNonRootUser,
+		HTTPProbe:       config.HTTPProbe,
+		SetNonRootUser:  config.SetNonRootUser,
 		ReadinessProbe: &k8s.ProbeConfig{
-			InitialDelaySeconds: int32(cfg.ReadinessProbeInitialDelaySeconds),
-			TimeoutSeconds:      int32(cfg.ReadinessProbeTimeoutSeconds),
-			PeriodSeconds:       int32(cfg.ReadinessProbePeriodSeconds),
+			InitialDelaySeconds: int32(config.ReadinessProbeInitialDelaySeconds),
+			TimeoutSeconds:      int32(config.ReadinessProbeTimeoutSeconds),
+			PeriodSeconds:       int32(config.ReadinessProbePeriodSeconds),
 		},
 		LivenessProbe: &k8s.ProbeConfig{
-			InitialDelaySeconds: int32(cfg.LivenessProbeInitialDelaySeconds),
-			TimeoutSeconds:      int32(cfg.LivenessProbeTimeoutSeconds),
-			PeriodSeconds:       int32(cfg.LivenessProbePeriodSeconds),
+			InitialDelaySeconds: int32(config.LivenessProbeInitialDelaySeconds),
+			TimeoutSeconds:      int32(config.LivenessProbeTimeoutSeconds),
+			PeriodSeconds:       int32(config.LivenessProbePeriodSeconds),
 		},
-		ImagePullPolicy: cfg.ImagePullPolicy,
+		ImagePullPolicy:   config.ImagePullPolicy,
+		ProfilesNamespace: config.ProfilesNamespace,
 	}
 
-	factory := k8s.NewFunctionFactory(clientset, deployConfig)
+	log.Printf("DeploymentConfig: %+v\n", deployConfig)
```

Comment:
Did you mean to leave this in?

Reply:
yes, i found it helpful to understand how the server was configured at startup

Reply:
It's not very user-friendly output:

```
ReadinessProbe:0xc000043d80 LivenessProbe:0xc000043d90 
```

I think the previous comment got removed, is that right? That printed the timeouts in a more verbose way?